src/encoded/static/components/report.js

```diff
@@ -282,53 +283,190 @@ ColumnHeader.propTypes = {
+            prevState.columns[columnPaths[0]].visible = true;
+            return { columns: prevState.columns };
+        });
+    }
+
+    // Called when the sorting option gets changed.
+    handleSortChange(sortOption) {
+        this.setState({ sortOption });
     }
 
     render() {
-        const columnPaths = Object.keys(this.props.columns);
+        const { columns } = this.state;
+        const firstColumnKey = Object.keys(columns)[0];
+
+        // Get the column paths, sorting them by the corresponding column title if the user asked
+        // for that.
+        let columnPaths = Object.keys(columns);
+        if (this.state.sortOption === 'alpha') {
+            columnPaths = columnPaths.sort((aKey, bKey) => {
+                const aTitle = columns[aKey].title.toLowerCase();
+                const bTitle = columns[bKey].title.toLowerCase();
+                return (aTitle < bTitle ? -1 : (bTitle < aTitle ? 1 : 0));
+            });
+        }
+
         return (
-            <div style={{ display: 'inline-block', position: 'relative' }}>
-                <button className={`btn btn-info btn-sm${this.state.open ? ' active' : ''}`} onClick={this.toggle} title="Choose columns">
-                    <i className="icon icon-columns" /> Columns
-                </button>
-                {this.state.open && <div style={{ position: 'absolute', top: '30px', width: '230px', backgroundColor: '#fff', padding: '.5em', border: 'solid 1px #ccc', borderRadius: 3, zIndex: 1 }}>
-                    <h4>Columns</h4>
-                    {columnPaths.map((columnPath) => {
-                        const column = this.props.columns[columnPath];
-                        return <ColumnItem key={columnPath} columnPath={columnPath} column={column} toggleColumn={this.toggleColumn} />;
-                    })}
-                </div>}
-            </div>
+            <Modal addClasses="column-selector">
+                <ModalHeader title="Select columns to view" closeModal={this.props.closeSelector} />
+                <ColumnSelectorControls
+                    handleSelectAll={this.handleSelectAll}
+                    handleSelectOne={this.handleSelectOne}
+                    handleSortChange={this.handleSortChange}
+                    firstColumnTitle={columns[firstColumnKey].title}
+                />
+                <ModalBody>
+                    <div className="column-selector__selectors">
+                        {columnPaths.map((columnPath) => {
+                            const column = columns[columnPath];
+                            return <ColumnItem key={columnPath} columnPath={columnPath} column={column} toggleColumn={this.toggleColumn} />;
+                        })}
+                    </div>
+                </ModalBody>
+                <ModalFooter
+                    closeModal={this.props.closeSelector}
+                    submitBtn={this.submitHandler}
+                    submitTitle="Select"
```

Comment:
Since the buttons at the top of the modal also say "Select", I think this one would be clearer with a different title; maybe "Update Report"

Reply:
Yeah, I agree.